src/providers/google/util.ts

```diff
@@ -704,35 +750,13 @@ export function geminiFormatAndSystemInstructions(
     contents = updatedContents;
   }
 
-  let systemInstruction: Content | string | undefined = parsedSystemInstruction;
-  if (configSystemInstruction && !systemInstruction) {
-    // Make a copy
-    systemInstruction = clone(configSystemInstruction);
-
-    // Load SI from file
-    if (typeof configSystemInstruction === 'string') {
-      systemInstruction = loadFile(configSystemInstruction, contextVars);
-    }
-
-    // Format SI if string was not a filepath above
-    if (typeof systemInstruction === 'string') {
-      systemInstruction = { parts: [{ text: systemInstruction }] };
-    }
+  let systemInstruction: Content | undefined = parsedSystemInstruction;
```

Comment:
Change type from `Content | string | undefined` to `Content | undefined`.
According to the type definition of `maybeCoerceToGeminiFormat`, `string` cannot be passed directly.